### PR TITLE
Rover: add new GCS failsafe timeout param and improve failsafe descri…

### PR DESCRIPTION
**Pull request by khancyr**
…ption

Pr for https://github.com/ArduPilot/ardupilot/pull/22794

### CODE REVIEW

File: rover/source/docs/rover-failsafes.rst [hunk 1]
```diff
@@ -69,11 +69,38 @@ Advanced Battery Failsafe Settings
 GCS Failsafe (aka Telemetry Failsafe)
 -------------------------------------
 
+Prior Rover-4.4
++++++++++++++++
+
 This failsafe is triggered if the vehicle stops receiving `heartbeat messages <https://mavlink.io/en/messages/common.html#HEARTBEAT>`__ from the ground station for at least :ref:`FS_TIMEOUT <FS_TIMEOUT>` seconds.
 
-- set :ref:`FS_GCS_ENABLE <FS_GCS_ENABLE>` to "1" to enable this failsafe
```
Comment (Hwurzburg) [inline rover/source/docs/rover-failsafes.rst:74]: dont remove these if you keep the prior to 4.4 section (which we should)

File: rover/source/docs/rover-failsafes.rst [hunk 2]
```diff
@@ -69,11 +69,38 @@ Advanced Battery Failsafe Settings
...
+Setting the failsafe
+++++++++++++++++++++
+
+In parameters list, set the :ref:`FS_GCS_ENABLE <FS_GCS_ENABLE>` parameter to:
+
+-  **Disabled** (Value 0) will disable the GCS failsafe entirely.
+-  **Enabled** (Value 1) will execute the FS_ACTION when Failsafe trigger.
+-  **Enabled Continue with Mission in Auto Mode** (Value 2) will ignore the failsafe in an Auto Mode mission.
+
+The action done on Failsafe is controled by :ref:`FS_ACTION <FS_ACTION>` parameter:
```
Comment (Hwurzburg) [inline rover/source/docs/rover-failsafes.rst:94]:
```suggestion
The action done on GCS Failsafe is controlled by the :ref:`FS_ACTION <FS_ACTION>` parameter:
```

File: rover/source/docs/rover-failsafes.rst [hunk 3]
```diff
@@ -69,11 +69,38 @@ Advanced Battery Failsafe Settings
...
-- you must use the transmitter's mode switch to re-take control of the vehicle in :ref:`Manual <manual-mode>` (or any other mode), or change modes via the GCS iself, if communication is re-established.
+Starting from Rover-4.4
++++++++++++++++++++++++
+
+Significant improvements have been made in Rover-4.4 and later for Ground Control Failsafe .
+
+The Ground Station Control (GCS) failsafe controls how Rover will behave if contact with the GCS is lost.
+The GCS failsafe monitors the time since the last MAVLink heartbeat from the GCS. If no heartbeat is received :ref:`FS_GCS_TIMEOUT <FS_GCS_TIMEOUT>` seconds (Default is 5 seconds), the GCS failsafe event will trigger based on your parameter settings. Note that if no GCS is ever connected, the GCS failsafe will remain inactive regardless of parameter settings.
+
+Setting the failsafe
```
Comment (Hwurzburg) [inline rover/source/docs/rover-failsafes.rst:85]:
```suggestion
Enabling the failsafe
```

check title under to match length

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

**Comment by Hwurzburg** [comment]
@khancyr poke...needs a few tweaks

File: rover/source/docs/rover-failsafes.rst (hunk 1 above)
Comment (khancyr) [reply rover/source/docs/rover-failsafes.rst:74]: That is just moved not removed.
It probably make it more readable if I add the 4.4 specific things at the end. Will update

**Comment by Hwurzburg** [comment]
@khancyr reminder...this needs updating...its part of 4,4 so I would like to merge it ASAP

**Comment by khancyr** [comment]
I will do tonight

**Comment by khancyr** [comment]
@Hwurzburg update done, good to go now!

**Comment by Hwurzburg** [comment]
thanks

**Pull request merged**